source/_posts/2017/2017-05-18-symfony33-di-symplify-deprecations.md

```diff
@@ -0,0 +1,178 @@
+
+Don't worry:
+ 
+- **all have better replacement** (I will show you bellow)
+- **you can still use them** (they just won't get any updates)
+
+
+Let's look on each package now:
+
+## [DefaultAutowire](https://github.com/DeprecatedPackages/DefaultAutowire)
+
+This package turned on autowiring by default for most of services.
+
+### Before
+
+All services were autowired.
+
+### After
+
+[Use `_defaults` section](/blog/2017/05/07/how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3/#1-let-s-add-code-defaults-code):
+
+```yaml
+# app/config/services.yml
+services:
+    _defaults:
+        autowire: true
+```
+
+All services in this config will be now autowired. You can use it in any `.yml` file.
+
+
+## [ServiceDefinitionDecorator](https://github.com/DeprecatedPackages/ServiceDefinitionDecorator)
+
+This packages helped with service tagging and method calls by type.
+
+
+### Before
+
+```yaml
+# app/config/config.yml
+decorator:
+    Symfony\Component\Console\Command\Command:
+        tags:
+            - { name: "console.command" }
+
+    AbstractEntityRepository:
+        calls:
+            - ["setEntityManager", ["@event_manager"]]
+```
+
+### After
+
+Use [`autoconfigure`](/blog/2017/05/07/how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3/#2-use-autoconfigure)
+and [`_instanceof`](/blog/2017/05/07/how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3/#5-use-code-instanceof-code)
+features.
+
+```yaml
+# app/config/services.yml
+services:
+    autoconfigure: true 
+
+    Symfony\Component\Console\Command\Command: ~
+
+    _instanceof:
+        AbstractEntityRepository:
+            calls:
+                - ["setEntityManager", ["@event_manager"]]
+```
+
+### Minitip
+
+You can also use `@required` annotation instead of `_instanceof` in this case, as **[shared by Kevin Bond](/blog/2017/05/07/how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3/#comment-3306767439)** - thank you Kevin!
+
+
+
+## [AutoServiceRegistration](https://github.com/DeprecatedPackages/AutoServiceRegistration)
+
+This package helped you to register multiple services at once with Finder.
+ 
+### Before
+
+```yaml
+# app/config/config.yml
+symplify_auto_service_registration:
+    directories_to_scan:
+        - %kernel.root_dir%
+    class_suffixes_to_seek:
+        - Controller
+```
+
+### After 
+
+Use [PSR-4 based service autodiscovery and registration](/blog/2017/05/07/how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3/#4-use-psr-4-based-service-autodiscovery-and-registration).
+
+```yaml
+# app/config/services.yml
+
+services:
+    App\: # no more manual registration of similar groups of services 
+        resource: ../❴Controller,Command,Subscriber❵
+```
+
+Tada!
+
+And just like that, you can drop 4 of Symplify bundles and use Symfony 3.3 happily and safely ever after.
+
+---
+
+That's it for today. In next post, I will show you, **how to deprecate package without letting anyone behind**.
```

Comment:
That´s "all" for today. I "the" next post I will show you how... (bez čárek stačí)